### PR TITLE
must give password to set db in cli

### PR DESCRIPTION
**Pull request by ailisp**
ISSUE: Currently we allow user to set up database in appliance console cli without given a password, which should be disallowed. This is part of BZ https://bugzilla.redhat.com/show_bug.cgi?id=1431815
Another part (disallow --internal without --dbdisk) is fixed in https://github.com/ManageIQ/manageiq-gems-pending/pull/277

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1431815#c2

\cc @yrudman @carbonin @gtanzillo 

@miq-bot add-label wip, bug

### CODE REVIEW

**Comment by ailisp** [comment]
@miq-bot remove-label wip

File: lib/gems/pending/appliance_console/cli.rb [hunk 1]
```diff
@@ -182,13 +182,18 @@ def run
 
     def set_db
       raise "No encryption key (v2_key) present" unless key_configuration.key_exist?
+      raise "Must provide a password to set database" unless password?
```
Comment (carbonin) [inline lib/gems/pending/appliance_console/cli.rb]: I think we can improve this sentence a bit ...

Maybe "A password is required to configure a database"

Comment (yrudman) [reply lib/gems/pending/appliance_console/cli.rb]: 👍

File: lib/gems/pending/appliance_console/cli.rb [hunk 2]
```diff
@@ -182,13 +182,18 @@ def run
...
         set_internal_db
       else
         set_external_db
       end
     end
 
+    def password?
+      options[:password] && !options[:password].strip.empty?
+    end
+
```
Comment (yrudman) [inline lib/gems/pending/appliance_console/cli.rb]: would below look more complicated or easer to read ? : 
`options[:password] ? !options[:password].strip.empty? : false`

Comment (carbonin) [reply lib/gems/pending/appliance_console/cli.rb]: I think it's fine the way it is.

**Comment by carbonin** [comment]
Looks good @ailisp. Just fix up the rubocop complaints and we should be good to go.

In this case you can either just add the spaces or (as the rest of this file seems to do), use `expect` with the `do ... end` block notation. I'm fine with either.

File: lib/gems/pending/appliance_console/cli.rb (hunk 2 above)
Comment (ailisp) [reply lib/gems/pending/appliance_console/cli.rb]: Thanks @yrudman and @carbonin . Now should be good.

**Pull request merged**